src/opencl/mod.rs

```diff
@@ -42,49 +48,15 @@ impl fmt::Display for Brand {
 }
 
 pub struct Buffer<T> {
-    buffer: ocl::Buffer<u8>,
+    buffer: opencl3::memory::Buffer<u8>,
+    length: usize,
     _phantom: std::marker::PhantomData<T>,
 }
 
 impl<T> Buffer<T> {
+    /// The number of bytes / size_of(T)
     pub fn length(&self) -> usize {
```

Comment:
can we pleeeeease change this to be `len`?

Reply:
I'd keep it for now, it will likely (I'm still testing) go away soon anyway. So instead of two breaking changes it will only be a single one.